src/mltcontroller.cpp

```diff
@@ -934,10 +935,14 @@ Producer* Controller::setupNewProducer(Producer* newProducer) const
                 }
                 filter.reset(newProducer->filter(i));
             }
-            return chain;
+            ret = chain;
         }
     }
-    return new Mlt::Producer(newProducer);
+    if (ret) {
+        ret = new Mlt::Producer(newProducer);
+    }
+    ret->set("length", qUtf8Printable(length));
```

Comment:
This causes a crash. If `ret` is NULL, it is not replaced with a new one, but the `ret->set()` crashes on NULL dereference.

Reply:
fixed in fe3f8355